src/panels/lovelace/ha-panel-lovelace.ts

```diff
@@ -184,13 +184,21 @@ class LovelacePanel extends hassLocalizeLitMixin(LitElement) {
         const { config, mode } = this.lovelace!;
         try {
           // Optimistic update
-          this._updateLovelace({ config: newConfig, mode: "storage" });
+          this._updateLovelace({
+            config: newConfig,
+            mode: "storage",
+            editMode: true,
+          });
           await saveConfig(this.hass!, newConfig);
         } catch (err) {
           // tslint:disable-next-line
           console.error(err);
           // Rollback the optimistic update
-          this._updateLovelace({ config, mode });
+          this._updateLovelace({
+            config,
+            mode,
+            editMode: false,
```

Comment:
You are changing a method that is used in various places with various requirements, to solve a problem with one of the places where it is used. That is wrong and will have unintended consequences. Instead, we should activate edit mode inside the "take control" code.